doc/contribute/bin_blobs.rst

```diff
@@ -0,0 +1,229 @@
+
+* Precompiled libraries: Hardware enablement libraries, distributed in
+  precompiled binary form, typically for SoC peripherals. An example could be an
+  enablement library for a wireless peripheral
+* Firmware images: An image containing the executable code for a secondary
+  processor or CPU.  This can be full or partial (typically delta or patch data)
+  and is generally copied into RAM or flash memory by the main CPU. An example
+  could be the firmware for the core running a Bluetooth LE Controller
+* Miscellaneous binary data files. An example could be pre-trained neural
+  network model data
+
+Hardware agnostic features provided via a proprietary library are not
+acceptable. For example, a proprietary and hardware agnostic TCP/IP stack
+distributed as a static archive would be rejected.
+
+Note that just because a blob has an acceptable type does not imply that it will
+be unconditionally accepted by the project; any blob may be rejected for other
+reasons on a case by case basis (see library-specific requirements below).
+In case of disagreement, the TSC is the arbiter of whether a particular blob
+fits in one of the above types.
+
+Precompiled library-specific requirements
+=========================================
+
+This section contains additional requirements specific to precompiled library
+blobs.
+
+Any person who wishes to submit a precompiled library must represent that it
+meets these requirements. The project may remove a blob from the upstream
+distribution if it is discovered that the blob fails to meet these requirements
+later on.
+
+Interface header files
+----------------------
+
+The precompiled library must be accompanied by one or more header files,
+distributed under a non-copyleft OSI approved license, that define the interface
+to the library.
+
+Allowed dependencies
+--------------------
+
+This section defines requirements related to external symbols that a library
+blob requires the build system to provide.
+
+* The blob must not depend on Zephyr APIs directly. In other words, it must have
+  been possible to build the binary without any Zephyr source code present at
+  all. This is required for loose coupling and maintainability, since Zephyr
+  APIs may change and such blobs cannot be modified by all project maintainers
+* Instead, if the code in the precompiled library requires functionality
+  provided by Zephyr (or an RTOS in general), an implementation of an OS
+  abstraction layer (aka porting layer) can be provided alongside the library.
+  The implementation of this OS abstraction layer must be in source code form,
+  released under an OSI approved license and thoroughfully documented using
+  Doxygen
+
+Toolchain requirements
+----------------------
+
+Precompiled library blobs must be in a data format which is compatible with and
+can be linked by a toolchain supported by the Zephyr Project (i.e. a toolchain
+that can build Zephyr applications). This is required for maintainability and
```

Comment:
This `i.e.` parenthetical is a non sequitur in my opinion.

Just because a toolchain can build zephyr applications does not mean that the toolchain is supported by the zephyr project itself. Oticon's toolchain is a counterexample.

Reply:
Correct you are. This was again a free adaptation of someone else's comment. I will rewrite this so that it reflects what we really mean here.